src/metafile.c

```diff
@@ -599,12 +599,19 @@ gdip_metafile_StretchDIBits (MetafilePlayContext *context, int XDest, int YDest,
 	printf ("\n\t\tClrImportant: %d", lpBitsInfo->bmiHeader.biClrImportant);
 #endif
 	ms.ptr = (BYTE*)lpBitsInfo;
-	ms.size = lpBitsInfo->bmiHeader.biSizeImage;
+	if (lpBitsInfo->bmiHeader.biCompression == 0) { // 0 == RGB 
+		// Per the spec, if compression is RGB ImageSize must be ignored (and it should be zero anyway)
+		// and calculated according to the following formula.
+		ms.size = (((lpBitsInfo->bmiHeader.biWidth * lpBitsInfo->bmiHeader.biPlanes * 
+			lpBitsInfo->bmiHeader.biBitCount + 31) & ~31) / 8) * abs(lpBitsInfo->bmiHeader.biHeight);
```

Comment:
the thing that is being multiplied by `abs(lpBits...)` needs to be floored according to https://en.m.wikipedia.org/wiki/BMP_file_format#Pixel_Storage